src/castling.rs

```diff
@@ -0,0 +1,127 @@
+#![allow(non_upper_case_globals)]
+
+use crate::board::Board;
+use crate::player::Player;
+use crate::pos::Pos;
+use crate::pos::*;
+use std::fmt;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct CastleAbility {
+    pub king: bool,
+    pub queen: bool,
```

Comment:
if I'm understanding this correctly, `king` and `queen` represent the castleside. Perhaps appending `side` to these variable names would make that more clear?

Reply:
A fun way to make this unnecessarily complicated could be to use something like [enum-map](https://gitlab.com/KonradBorowski/enum-map).

Reply:
5494a586141cb960da37014f9c3a1ee3dc72cf75 ... enum-map looks interesting but i kept it simple for this